internal/http/server.go

```diff
@@ -127,7 +127,7 @@ func httpForCode(c codes.Code) int {
 		return http.StatusConflict
 	case codes.NotFound:
 		return http.StatusNotFound
-	case codes.FailedPrecondition, codes.InvalidArgument:
+	case codes.FailedPrecondition, codes.InvalidArgument, codes.Unauthenticated:
```

Comment:
Unauthenticated being mapped to BadRequest is surprising. I would expect http.Forbidden or http.Unauthorized. It successfully works around the change you've made in this PR without breaking existing functionality, but I think it's the kind of spaghetti bear trap that will catch us out in the future if we have need to use authentication for any witness operations.

Reply:
With this change, `Unauthenticated` means "Invalid Proof", so `Bad Request` doesn't seem super surprising in that context? 

I agree about the complexity, that's why I mention in the description that the better fix would be to bring the witness API more into line with how the world looks nowadays.

This PR was meant to sketch out a plausible workaround in the meantime.

Reply:
Sure, but the docs for `Unauthenticated` say `The request does not have valid authentication credentials for the operation.` which seems like a different situation. We can get into redefining a consistent proof to be a form of authentication credential, but that's getting into bonkers territory :-)

If this is a temporary change while we do the bigger refactor of the internal API and we add appropriate TODOs then I'm fine with this as a pragmatic step. I hope we can both agree that leaving the code like this makes maintainability much more difficult.